src/arrayMethodJoin.js

```diff
@@ -5,7 +5,22 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let result = '';
+    const mySeparator = separator === undefined ? ',' : separator;
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] === null || this[i] === '' || this[i] === undefined) {
```

Comment:
I think you don't need this check: `this[i] === ''` :)

Reply:
agree, fix @sv-dubov comment